ironmon_tracker/screens/CustomExtensionsScreen.lua

```diff
@@ -0,0 +1,262 @@
+		onClick = function(self)
+			CustomExtensionsScreen.resetButtons()
+			Program.changeScreenView(Program.Screens.SETUP)
+		end
+	},
+}
+
+function CustomExtensionsScreen.initialize()
+	for _, button in pairs(CustomExtensionsScreen.Buttons) do
+		if button.textColor == nil then
+			button.textColor = CustomExtensionsScreen.Colors.text
+		end
+		if button.boxColors == nil then
+			button.boxColors = { CustomExtensionsScreen.Colors.border, CustomExtensionsScreen.Colors.boxFill }
+		end
+	end
+	CustomExtensionsScreen.Buttons.EnableCustomExtensions.toggleState = Options["Enable custom extensions"]
+	CustomExtensionsScreen.Buttons.CurrentPage:updateText()
+end
+
+function CustomExtensionsScreen.refreshExtensionList()
+	-- Used to help remove any inactive or missing extension files
+	local activeExtensions = {}
+
+	local customFolderPath = FileManager.getCustomFolderPath()
+	local customFiles = FileManager.getFilesFromDirectory(customFolderPath)
+	for _, filename in pairs(customFiles) do
+		local name = FileManager.extractFileNameFromPath(filename) or ""
+		local ext = FileManager.extractFileExtensionFromPath(filename) or ""
+		ext = "." .. ext
+
+		-- Load any new Lua code files, but only if they don't already exist
+		if ext == FileManager.Extensions.LUA_CODE then
+			if CustomCode.ExtensionLibrary[name] == nil then
+				CustomCode.loadExtension(name)
+			end
+			activeExtensions[name] = true
+		end
+	end
+
+	for extensionKey, _ in pairs(CustomCode.ExtensionLibrary) do
+		if not activeExtensions[extensionKey] then
+			CustomCode.disableExtension(extensionKey)
+			CustomCode.ExtensionLibrary[extensionKey] = nil
+			Main.RemoveMetaSetting("extensions", extensionKey)
+		end
+	end
+
+	Main.SaveSettings(true)
+end
+
+function CustomExtensionsScreen.buildOutPagedButtons()
+	CustomExtensionsScreen.Pager.Buttons = {}
+
+	for extensionKey, extension in pairs(CustomCode.ExtensionLibrary) do
+		local image = Utils.inlineIf(extension.isEnabled, Constants.PixelImages.CHECKMARK, Constants.PixelImages.CROSS)
+		local iconColors = Utils.inlineIf(extension.isEnabled, { "Positive text", }, { "Negative text", })
+		local button = {
+			type = Constants.ButtonTypes.ICON_BORDER,
+			image = image,
+			text = extension.name or Constants.BLANKLINE,
+			textColor = CustomExtensionsScreen.Colors.text,
+			iconColors = iconColors,
+			extension = extension,
+			extensionKey = extensionKey,
+			dimensions = { width = 132, height = 16, },
+			isVisible = function(self) return CustomExtensionsScreen.Pager.currentPage == self.pageVisible end,
+			updateText = function(self)
+				if self.extension.isEnabled then
+					self.image = Constants.PixelImages.CHECKMARK
+					self.iconColors = { "Positive text", }
+				else
+					self.image = Constants.PixelImages.CROSS
+					self.iconColors = { "Negative text", }
+				end
+			end,
+			onClick = function(self)
+				-- TODO: Probably navigate to single screen to show enable/disable options; for now, toggle it
```

Comment:
an idea on top of this maybe, could have a setup where clicking the extension name (or perhaps a button next to it) shows you the description / version number of the extension? (either a separate page or like some form of drop-down style box). Would need to enforce a max length for the description of course (or truncate it)